consume.go

```diff
@@ -239,17 +212,17 @@ func (cmd *consumeCmd) newClient() (sarama.Client, error) {
 	return client, nil
 }
 
-func (cmd *consumeCmd) consume(partitions []int32) error {
+func (cmd *consumeCmd) consume(partitions map[int32]resolvedInterval) error {
 	var wg sync.WaitGroup
 	out := make(chan printContext)
 	go print(out, cmd.pretty)
 
 	wg.Add(len(partitions))
-	for _, p := range partitions {
-		p := p
+	for p, interval := range partitions {
+		p, interval := p, interval
```

Comment:
WTF?!??!  :-)  I see why you do this, but it's worthy of a comment.   The number of times people fall into this gotcha makes not falling into it "surprising" ;-)

Reply:
actually, I might even use different names for the variables to, just to make it clear.

Reply:
I consider this to be a useful idiom - it comes up all the time, so I'm not sure it's worth having a comment It's sufficiently surprising in appearance that it should at least invite the question the first time it's seen, and be recognised for what it is thereafter.

IMHO It's a Good Thing to use the same names for the variables because it means that it's not actually possible to use the outer-scoped names incorrectly. And also because the only reason to use the `x := x` idiom is precisely for this reason - if we were using another name, then we'd have to look more closely at the code to see what we're doing.

Reply:
Yes, it's surprising, but still, I err on the side of explaining why I do something surprising.